modules/connectivity/locals.tf

```diff
@@ -930,14 +962,23 @@ locals {
           hub_network.config.azure_firewall.config.sku_tier,
           "Standard"
         )
-        firewall_policy_id          = try(local.custom_settings.azurerm_firewall["connectivity"][location].firewall_policy_id, local.azfw_policy_resource_id[location])
-        dns_servers                 = try(local.custom_settings.azurerm_firewall["connectivity"][location].dns_servers, null)
-        private_ip_ranges           = try(local.custom_settings.azurerm_firewall["connectivity"][location].private_ip_ranges, null)
-        management_ip_configuration = try(local.custom_settings.azurerm_firewall["connectivity"][location].management_ip_configuration, local.empty_list)
-        threat_intel_mode           = try(local.custom_settings.azurerm_firewall["connectivity"][location].threat_intel_mode, coalesce(hub_network.config.azure_firewall.config.threat_intelligence_mode, "Alert"))
-        virtual_hub                 = local.empty_list
-        zones                       = try(local.custom_settings.azurerm_firewall["connectivity"][location].zones, local.azfw_zones[location])
-        tags                        = try(local.custom_settings.azurerm_firewall["connectivity"][location].tags, local.tags)
+        firewall_policy_id = try(local.custom_settings.azurerm_firewall["connectivity"][location].firewall_policy_id, local.azfw_policy_resource_id[location])
+        dns_servers        = try(local.custom_settings.azurerm_firewall["connectivity"][location].dns_servers, null)
+        private_ip_ranges  = try(local.custom_settings.azurerm_firewall["connectivity"][location].private_ip_ranges, null)
+        management_ip_configuration = try(local.custom_settings.azurerm_firewall["connectivity"][location].management_ip_configuration,
+          hub_network.config.azure_firewall.config.sku_tier == "Basic" ?
```

Comment:
Is it mandatory for a 'Basic' sku firewall to have a public IP? Just trying to understand the reason for this part.

Reply:
Hey Jared, for the `management_ip_configuration` it is a mandatory setting to have a public ip if you are creating the AzureManagementFirewallSubnet, which is necessary for the deployment of the basic SKU. 

https://learn.microsoft.com/en-us/azure/firewall/deploy-firewall-basic-portal-policy